Use newer ffmpeg APIs to avoid deprecation warnings.

comex opened the pull request:
This should work even on the ancient (early 2013 and earlier) versions of ffmpeg that are still supported...

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dolphin-emu/dolphin/3946)
<!-- Reviewable:end -->

lioncash commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[Source/Core/VideoCommon/AVIDump.cpp, line 170 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3946#-KLALs6Vdx2NfkOc1EM0:-KLALs6Vdx2NfkOc1EM1:-1852432528) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ce1382ec368fc9aaa236bdc0947077a2243b9055/Source/Core/VideoCommon/AVIDump.cpp#L170)):*
> ```cpp
> void AVIDump::AddFrame(const u8* data, int width, int height)
> {
>   s_src_frame->data[0] = (uint8_t*) data;
> ```

Why was const_cast removed?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3946)*
<!-- Sent from Reviewable.io -->

comex commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[Source/Core/VideoCommon/AVIDump.cpp, line 170 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3946#-KLALs6Vdx2NfkOc1EM0:-KLANF8RneNdueyjcyVa:-89882681) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ce1382ec368fc9aaa236bdc0947077a2243b9055/Source/Core/VideoCommon/AVIDump.cpp#L170)):*
<details><summary><i>Previously, lioncash (Mat M.) wrote…</i></summary>
> Why was const_cast removed?
</details>
because I deleted the original version before writing that.  Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3946)*
<!-- Sent from Reviewable.io -->

RisingFog commented:
@comex This PR needs to be rebased.

RisingFog commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3946#-:-KMMg7-RKBhm01oFvRcp:1417088721)*
<!-- Sent from Reviewable.io -->
